### PR TITLE
Extending get/config endpoint to return idpList configuration

### PR DESCRIPTION
**Pull request by dc-bcgov**
# Description

Extending 'get/config' endpoint to return appropriate 'idpList' configuration, and append it to the 'frontend' config. Default will be used if override does not exist. File exist check is covered through import/catch, though if default is not defined error will be thrown.

## Checklist

- [ ] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [ ] I have checked that unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/app.ts [hunk 1]
```diff
@@ -12,10 +12,12 @@ app.get('/', (req, res) => {
...
 // Frontend configuration endpoint
-app.get('/config', (_req, res, next) => {
+app.get('/config', async (_req, res, next) => {
   try {
-    const frontend = config.get('frontend');
-    res.status(200).json(frontend);
+    res.status(200).json({
+      ...config.get('frontend'),
+      idpList: await readIdpList()
+   });
```
Comment (jujaga) [inline app/app.ts:20]: Fix indentation (missing a space).

**Review by jujaga — APPROVED** [review]

File: app/config/idplist-default.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+[
+  {
+    "elevatedRights": true,
```
Comment (TimCsaky) [inline app/config/idplist-default.json:3]: wonder if `level: <int>` would work better (?)

File: app/app.ts [hunk 3]
```diff
@@ -28,4 +30,12 @@ app.use('/favicon.ico', (_req, res) => {
 });
 app.use(staticFilesPath, express.static(path.join(__dirname, 'dist')));
 
+async function readIdpList(): Promise<string> {
+  const idplistDefaultPath = './config/idplist-default.json';
+  const idplistOverridePath = './config/idplist-override.json';
```
Comment (TimCsaky) [inline app/app.ts:35]: would anyone prefer if overrides were in `idplist-local.json`

Comment (jujaga) [reply app/app.ts:35]: Could change it to be named local instead - would match better with existing config library patterns. 👍

File: app/config/idplist-default.json (hunk 2 above)
Comment (jujaga) [reply app/config/idplist-default.json:3]: We don't really have multi-tiered levels of access so I don't think we should go that route yet; a boolean suffices.

**Review by kamorel — APPROVED** [review]

**Pull request merged**